### PR TITLE
Update DoE Factorial

### PR DESCRIPTION
**Pull request by knrds**
Comment out bits that lack functionality
Edit DESCRIPTION to remove error

### CODE REVIEW

File: R/doeFactorial.R [hunk 1]
```diff
@@ -110,9 +110,9 @@ doeFactorial <- function(jaspResults, dataset, options, ...){
                        factors = options[["numberOfFactors"]],
                        runs = runs,
                        resolution = resolution,
-                       centers = options[["factorialCenterPoints"]],
-                       replicates = options[["factorialCornerReplicates"]],
-                       blocks = options[["factorialBlocks"]])
+                       centers = "NA",
+                       replicates = "NA",
+                       blocks = "NA")
```
Comment (vandenman) [inline R/doeFactorial.R:115]: as long as the default values don't crash, you only need to hide it in the QML file.

File: inst/qml/doeFactorial.qml [hunk 2]
```diff
@@ -18,206 +18,206 @@ import QtQuick.Layouts 							1.3
...
-		values:
-		[
-			{ value: "2", label: qsTr("2")},
-            { value: "3", label: qsTr("3")},
-            { value: "Mixed", label: qsTr("Mixed")}
-		]
-		}
-	}
+    usesJaspResults:							true
+    columns:									2
```
Comment (vandenman) [inline inst/qml/doeFactorial.qml:24]: 1. If you add `debug: true` or `visible: false` you don't need to comment out all of the code.

2. Please use tabs, not spaces. See https://doc.qt.io/qtcreator/creator-indenting-code.html#indenting-qml-files for how to change the settings in Qt.

**Review by vandenman — CHANGES_REQUESTED** [review]
Is this PR related to #65? Because the bugs are still there.

File: R/doeFactorial.R (hunk 1 above)
Comment (knrds) [reply R/doeFactorial.R]: the default values are inaccurate with some of the input options, so I've hidden them for now

File: inst/qml/doeFactorial.qml (hunk 2 above)
Comment (knrds) [reply inst/qml/doeFactorial.qml]: done!

**Comment by knrds** [comment]
1. I fixed the design resolution bug where a value from the hard coded table would be taken. Now resolution (and runs) are calculated in R
2. I don't understand this error, because when I click on 'Preview design' everything works fine
3. A factorial design with 4 factors and only 4 runs is impossible, so it makes sense that the user should get an error, no?

File: R/doeFactorial.R [hunk 3]
```diff
@@ -92,27 +92,36 @@ doeFactorial <- function(jaspResults, dataset, options, ...){
     designs <- jaspResults[["state"]]$object
 
     if(options[["designBy"]] == "designByRuns"){
-      rowIndex 	    <- which(designs[, 1] == options[["factorialRuns"]])
-      resolution    <- designs[rowIndex, options[["numberOfFactors"]]]
-      runs          <- options[["factorialRuns"]]
+      runs <- options[["factorialRuns"]]
+      if(log2(as.numeric(options[["factorialRuns"]])) < options[["numberOfFactors"]]){
+      resolution <- DoE.base::design.info(FrF2::FrF2(nfactors = as.numeric(options[["numberOfFactors"]]),
```
Comment (vandenman) [inline R/doeFactorial.R:97]: if you add a package you should also add it to the DESCRIPTION file.

File: inst/qml/doeFactorial.qml [hunk 4]
```diff
@@ -18,40 +18,40 @@ import QtQuick.Layouts 							1.3
...
-		indexDefaultValue: 						0
-		values:
-		[
-			{ value: "2", label: qsTr("2")},
-            { value: "3", label: qsTr("3")},
-            { value: "Mixed", label: qsTr("Mixed")}
-		]
-		}
-	}
+    usesJaspResults:                            true
```
Comment (vandenman) [inline inst/qml/doeFactorial.qml:23]: these are still all spaces instead of tabs

**Review by vandenman — DISMISSED** [review]
Some comments, I'll test it out locally to see if I can still replicate 2.

File: inst/qml/doeFactorial.qml (hunk 4 above)
Comment (knrds) [reply inst/qml/doeFactorial.qml:23]: is there a way I can easily view if I did tabs or spaces?

File: R/doeFactorial.R (hunk 3 above)
Comment (knrds) [reply R/doeFactorial.R:97]: ah okay, I thought that since DoE.base is a dependency of FrF2, I didn't have to add it, will do so now

**Comment by vandenman** [comment]
I can still reproduce issue 2 but if it doesn't happen on macOS/ only happens on Ubuntu we can ignore it for now.

File: R/doeFactorial.R (hunk 3 above)
Comment (vandenman) [reply R/doeFactorial.R:97]: in that case, it would have probably worked. But that's not how it's supposed to be.

**Comment by vandenman** [comment]
> A factorial design with 4 factors and only 4 runs is impossible, so it makes sense that the user should get an error, no?

Yes, but you should throw an expected error, and not let R crash.

**Review by koenderks — APPROVED** [review]

**Pull request merged**